src/libcore/panicking.rs

```diff
@@ -39,9 +39,15 @@
 use fmt;
 use panic::{Location, PanicInfo};
 
-#[cold] #[inline(never)] // this is the slow path, always
+#[cold]
+// inline(never) is required even in panic_immediate_abort mode, lest linker error
+#[inline(never)]
 #[lang = "panic"]
 pub fn panic(expr_file_line_col: &(&'static str, &'static str, u32, u32)) -> ! {
+    if cfg!(feature = "panic_immediate_abort") {
+        unsafe { super::intrinsics::abort() }
+    };
```

Comment:
stray semicolon?

Reply:
I'm not sure about code style recommendations in case of `()`-returning `if` when called function does not return `()`. Where shall the semicolon go? As the result of `if` itself (like in this snippet) or inside the function? Or both? Or none?

Rustfmt seems to preserve both semicolons in

```rust
pub fn main() {
    if false {
        ();
    };
}
```

, so I assumed they should stay to remind it's not a part of larger expression, just a side-effects-full call.

Reply:
I think the semicolon can just be removed, right? Typically semicolon after `if` indicates an assignment, but there's no assignment here